### ISSUE TITLE
Unable to use cached boot2docker.iso in Windows

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#221]
Hi,

Unable to use cached boot2docker.iso in Windows 10 for both HyperV and VirtualBox  5.0.30 drivers.
And, for HyperV, it is directory communicating to github url for ISO url. 

I need to start with `--iso-url` option as `minishift.exe start --vm-driver virtualbox --iso-url file://C:/boot2docker.iso`.

### VirtualBox
```
PS D:\> ls C:\Users\budhram\.minishift\cache\iso\
    Directory: C:\Users\budhram\.minishift\cache\iso
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       12/29/2016   7:37 PM       37748736 boot2docker.iso

PS D:\> .\minishift.exe version
minishift version: 1.0.0-beta.1

PS D:\> .\minishift.exe start --vm-driver virtualbox -v 5 --show-libmachine-logs
E1230 13:54:23.197297    5848 update.go:76] Get https://api.github.com/repos/jimmidyson/minishift/releases/latest: dial tcp: lookup api.github.com: getaddrinfow: No such host is known.
Starting local OpenShift cluster...
Found binary path at D:\minishift.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:49698
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetMachineName
(minishift) Calling .DriverName
Running pre-create checks...
(minishift) Calling .PreCreateCheck
(minishift) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe --version
(minishift) DBG | STDOUT:
(minishift) DBG | {
(minishift) DBG | 5.0.30r112061
(minishift) DBG | }
(minishift) DBG | STDERR:
(minishift) DBG | {
(minishift) DBG | }
(minishift) DBG | Hyper-V is not installed.
(minishift) DBG | %!(EXTRA *exec.Error=exec: "vmms.exe": executable file not found in %PATH%)COMMAND: wmic cpu get VirtualizationFirmwareEnabled
(minishift) DBG | STDOUT:
(minishift) DBG | {
(minishift) DBG | VirtualizationFirmwareEnabled
(minishift) DBG | TRUE
(minishift) DBG |
(minishift) DBG | }
(minishift) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe list hostonlyifs
(minishift) DBG | STDOUT:
(minishift) DBG | {
(minishift) DBG | Name:            VirtualBox Host-Only Ethernet Adapter #2
(minishift) DBG | GUID:            86707bde-8f13-4f5d-8381-f953689021cf
(minishift) DBG | DHCP:            Disabled
(minishift) DBG | IPAddress:       192.168.99.1
(minishift) DBG | NetworkMask:     255.255.255.0
(minishift) Calling .GetConfigRaw
(minishift) DBG | IPV6Address:
(minishift) DBG | IPV6NetworkMaskPrefixLength: 0
(minishift) DBG | HardwareAddress: 0a:00:27:00:00:0c
(minishift) DBG | MediumType:      Ethernet
(minishift) DBG | Status:          Up
(minishift) DBG | VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter #2
(minishift) DBG |
(minishift) DBG | }
Creating machine...
(minishift) DBG | STDERR:
(minishift) DBG | {
(minishift) Calling .Create
(minishift) DBG | }
(minishift) Downloading C:\Users\budhram\.minishift\cache\boot2docker.iso from file://C:/Users/budhram/.minishift/cache/iso/boot2docker.iso...
E1230 13:54:26.113439    5848 start.go:130] Error starting host: Error creating. Error creating machine: Error in driver during machine creation: open /Users/budhram/.minishift/cache/iso/boot2
docker.iso: The system cannot find the path specified.. Retrying.
Found binary path at D:\minishift.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:49702
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe showvminfo minishift --machinereadable
(minishift) DBG | STDOUT:
(minishift) DBG | {
E1230 13:54:26.300224    5848 start.go:130] Error starting host: Error getting state for host: machine does not exist. Retrying.
(minishift) DBG | }
```
### HyperV
```
Found binary path at D:\minishift.exe
Launching plugin server for driver hyperv
Plugin server listening at address 127.0.0.1:49689
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] : Get-VM : Hyper-V was unable to find a virtual machine with name "minishift".
(minishift) Calling .Start
(minishift) DBG | At line:1 char:3
(minishift) DBG | + ( Get-VM minishift ).state
(minishift) DBG | +   ~~~~~~~~~~~~~~~~
(minishift) DBG |     + CategoryInfo          : InvalidArgument: (minishift:String) [Get-VM], VirtualizationException
(minishift) DBG |     + FullyQualifiedErrorId : InvalidParameter,Microsoft.HyperV.PowerShell.Commands.GetVM
(minishift) DBG |
(minishift) DBG |
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Start-VM minishift
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] : Start-VM : Hyper-V was unable to find a virtual machine with name "minishift".
E1230 14:05:42.876882    6336 start.go:130] Error starting host: Error starting stopped host: exit status 1. Retrying.
(minishift) DBG | At line:1 char:1
(minishift) DBG | + Start-VM minishift
(minishift) DBG | + ~~~~~~~~~~~~~~~~~~
(minishift) DBG |     + CategoryInfo          : ObjectNotFound: (:) [Start-VM], VirtualizationException
Found binary path at D:\minishift.exe
(minishift) DBG |     + FullyQualifiedErrorId : ObjectNotFound,Microsoft.HyperV.PowerShell.Commands.StartVM
(minishift) DBG |
(minishift) DBG |
Launching plugin server for driver hyperv
Plugin server listening at address 127.0.0.1:49693
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetState
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive ( Get-VM minishift ).state
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] : Get-VM : Hyper-V was unable to find a virtual machine with name "minishift".
(minishift) Calling .Start
(minishift) DBG | At line:1 char:3
(minishift) DBG | + ( Get-VM minishift ).state
(minishift) DBG | +   ~~~~~~~~~~~~~~~~
(minishift) DBG |     + CategoryInfo          : InvalidArgument: (minishift:String) [Get-VM], VirtualizationException
(minishift) DBG |     + FullyQualifiedErrorId : InvalidParameter,Microsoft.HyperV.PowerShell.Commands.GetVM
(minishift) DBG |
(minishift) DBG |
(minishift) DBG | [executing ==>] : C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Start-VM minishift
(minishift) DBG | [stdout =====>] :
(minishift) DBG | [stderr =====>] : Start-VM : Hyper-V was unable to find a virtual machine with name "minishift".
E1230 14:05:43.904542    6336 start.go:130] Error starting host: Error starting stopped host: exit status 1. Retrying.
(minishift) DBG | At line:1 char:1
E1230 14:05:43.907476    6336 start.go:136] Error starting host:  Error creating. Error creating machine: Error in driver during machine creation: Get https://github.com/
minishift/releases/download/v1.0.0-beta.1/boot2docker.iso: dial tcp: lookup github.com: getaddrinfow: No such host is known.
Error starting stopped host: exit status 1
Error starting stopped host: exit status 1
(minishift) DBG | + Start-VM minishift
```

**Comment by hferentschik** [comment]
The path is wrong 'open /Users/budhram/.minishift/cache/iso/boot2docker.iso: The system cannot find the path specified.. Retrying.' I have a feeling this is related to issue #206. I think we need to investigate whether we handle the users home directory properly and whether Go truly out of the box handles this properly. I have a feeling we might need to do some additional work there.

**Comment by hferentschik** [comment]
I think what we need here are some simple production code independent unit tests which deal with file operations based on the users home directory. With our current CI setup we can then already confirm that we have the expected results for OS X / Linux vs Windows.

@budhrg any chance Cygwin is involved, aka you are using cygwin terminals?

**Comment by budhrg** [comment]
No @hferentschik . Not used cygwin yet to test anything. Probably, I can try but I don't think we should deep dive into it more as most of things are working properly with PowerShell or comand prompt.

**Comment by praveenkumar** [comment]
@budhrg As per log looks like `file://C:/Users/budhram/.minishift/cache/iso/boot2docker.iso` is not accessible after downloading this iso can you directly use this path with `--iso-url` and check if we find real issue.

**Comment by budhrg** [comment]
>  can you directly use this path with --iso-url and check if we find real issue.

Yes, `--iso-url` option is working fine with format `file://<Drive Letter>:/<iso path>`.

**Comment by budhrg** [comment]
It seems that this issue only exits in VirtualBox driver only.

I am able to start, deploy application in HyperV.

**Comment by budhrg** [comment]
Hi, I have updated the issue title and description as per my new findings.

**Comment by budhrg** [comment]
Finally, found the cause of the issue. i.e I was using the `minishift.exe` binary from `D:` drive which is 
not my system / user profile drive. Created issue https://github.com/minishift/minishift/issues/236 to track it. 

Actually, the caching related issue is not for VirtualBox provider.

It was there in HyperV and added its fix in the PR #237 .

Also, documented same under the installation instructions.

**Comment by hferentschik** [comment]
Merged by pull request #237
